framework/include/constraints/MortarConstraintBase.h

```diff
@@ -85,6 +85,31 @@ class MortarConstraintBase : public Constraint,
    */
   void setNormals(const std::vector<Point> & normals) { _normals = normals; }
 
+  /**
+   * Set the nodal tangents
+   */
+  void setNodalTangents(const std::array<std::vector<Point>, 2> & tangents)
+  {
+    _tangents_3d = tangents;
+  }
```

Comment:
I don't know what to make of the doxygen and method name seemingly being generic and dimension agnostic, but the data member name specifying 3d. I feel like something should change here but I don't know what. Should this method only be called in 3d and hence perhaps the method name should be changed? Or should the data member name be changed?

Reply:
These are nodal tangents obtained from an orthogonalization process starting with the interpolated nodal normal vectors. I think I added 3d to make sure I was using those for 3d applications, since it guarantees that they are geometrically and 'mortarly' consistent. But I think we can call them `_nodal_tangents` which will distinguish them from the `tangents` coming from `libmesh`.

Reply:
I would like that